### ISSUE TITLE
BasemapGallery - Clone basemap prior to setting to avoid "object already owned" error.

### ISSUE THREAD

**Issue by mhdostal** [Esri/arcgis-maps-sdk-swift-toolkit#328]
When the client passes in a list of base maps, and if multiple web maps are used with the BasemapGallery, you can get the "object already owned error". The solution is to clone the basemap prior to setting the new basemap.

Cloning is NOT necessary when the user passes in a portal, as the portal basemaps are reloaded every time the basemap gallery is displayed.

**Comment by dfeinzimer** [comment]
@mhdostal I'm trying to repro this to adjust the fix for #1126 but I'm not understanding the description:

> and if multiple web maps are used with the BasemapGallery

Do you recall what the steps to reproduce were?

**Comment by mhdostal** [comment]
In `BasemapGalleryExampleView`, create a second map property:
```
    @State private var map2: Map = {
        let map = Map(basemapStyle: .arcGISImagery)
        map.initialViewpoint = Viewpoint(
            center: Point(x: -93.258133, y: 44.986656, spatialReference: .wgs84),
            scale: 1_000_000
        )
        return map
    }()
```

Then, create a `.task` on the `MapView` in the body of the example:

```
            .task {
                map2.basemap = basemaps.first?.basemap
                try? await map2.load()
            }
```

Then run the example and set the basemap of the original map to the first basemap in the list. If there is no `.clone()` call, you get the `Object already owned` error.
